cmd/traffic_manager/traffic_manager.cc

```diff
@@ -740,16 +742,17 @@ main(int argc, const char **argv)
     }
 
     if (lmgmt->run_proxy && !lmgmt->processRunning() && lmgmt->proxy_recoverable) { /* Make sure we still have a proxy up */
-      if (sleep_time) {
+      const uint64_t now = static_cast<uint64_t>(time(nullptr));
+      if (sleep_time && ((now - last_start_epoc_s) < MAX_SLEEP_S)) {
```

Comment:
I'm not sure about this logic. If the last sleep was less than `MAX_SLEEP_S` then it's reset to 1? What if the last sleep time was `MAX_SLEEP_S` - won't that mean cycling back to 1ms? I would think `sleep` should be reset once the proxy has fired up successfully and not before then.

Reply:
This is testing if the last start—not the last sleep—was less that `MAX_SLEEP_S` ago.

If it has not been up at least `MAX_SLEEP_S`, then we want to continue our exponential backoff from where we left it.

Otherwise, the server was last started at least `MAX_SLEEP_S` ago, in which case we should reset our sleep to the beginning.